processing/src/main/java/org/apache/druid/segment/join/HashJoinSegmentStorageAdapter.java

```diff
@@ -297,6 +302,22 @@ public boolean isEnableFilterPushDown()
     return enableFilterPushDown;
   }
 
+  @VisibleForTesting
```

Comment:
nit: public functions marked `VisibleForTesting` are hard to maintain, because people can accidentally call this function from another package and there's nothing that warns them they're doing something wrong.

I also noticed this function is only called in a test, can we refactor this so that we don't need a public VisibleForTesting function

Reply:
I collapsed this into a shared method with the block in makeCursors and removed `VisibleForTesting`